algorithms/linfa-clustering/src/dbscan/algorithm.rs

```diff
@@ -142,62 +202,27 @@ impl<F: Float, D: Data<Elem = F>, T>
     }
 }
 
-impl<F: Float, D: Data<Elem = F>> Transformer<&ArrayBase<D, Ix2>, Array1<Option<usize>>>
-    for DbscanHyperParamsBuilder<F>
-{
-    fn transform(&self, observations: &ArrayBase<D, Ix2>) -> Array1<Option<usize>> {
+impl<F: Float, DF: 'static + Distance<F>, N: NearestNeighbour> DbscanHyperParamsBuilder<F, DF, N> {
+    pub fn transform<D: Data<Elem = F>>(
```

Comment:
both of these transforms are actually prediction, you only have to implement `predict_ref` here. `Transform` would take the variables and transform them into integer associations